src/docgen/view/documentation.ts

```diff
@@ -145,78 +147,92 @@ export class Documentation {
-      switch (options?.language ?? Language.TYPESCRIPT) {
-        case Language.PYTHON:
-          language = TargetLanguage.PYTHON;
-          transpile = new PythonTranspile();
-          break;
-        case Language.TYPESCRIPT:
-          transpile = new TypeScriptTranspile();
-          break;
-        case Language.JAVA:
-          language = TargetLanguage.JAVA;
-          transpile = new JavaTranspile();
-          break;
-        case Language.CSHARP:
-          language = TargetLanguage.CSHARP;
-          transpile = new CSharpTranspile();
-          break;
-        default:
-          throw new Error(`Unsupported language: ${options?.language}. Supported languages are ${Object.values(Language)}`);
-      }
-      const assembly = await createAssembly(assemblyName, workdir, options?.loose ?? true, language);
-      return new Documentation(assembly, transpile);
-    });
-
+  public static async forAssembly(assemblyName: string, assembliesDir: string): Promise<Documentation> {
+    return new Documentation(assemblyName, assembliesDir);
   }
 
+  private readonly cleanupDirectories: string[] = [];
+  private readonly assembliesCache: Map<string, reflect.Assembly> = new Map();
+
   private constructor(
-    public readonly assembly: reflect.Assembly,
-    private readonly transpile: Transpile,
-  ) {
-  }
+    private readonly assemblyName: string,
+    private readonly assembliesDir: string,
+  ) {}
 
   /**
    * Generate markdown.
    */
-  public render(options?: RenderOptions): Markdown {
-    const submodule = options?.submodule ? this.findSubmodule(this.assembly, options.submodule) : undefined;
+  public async render(options?: RenderOptions): Promise<Markdown> {
+
+    const language = options?.language ?? Language.TYPESCRIPT;
+    const loose = options?.loose ?? true;
+
+    const { assembly, transpile } = await this.languageSpecific(language, loose);
+
+    const submodule = options?.submodule ? this.findSubmodule(assembly, options.submodule) : undefined;
     const documentation = new Markdown();
 
     if (options?.readme ?? true) {
-      const readme = new Readme(this.transpile, this.assembly, submodule);
+      const readme = new Readme(transpile, assembly, submodule);
       documentation.section(readme.render());
     }
 
     if (options?.apiReference ?? true) {
-      const apiReference = new ApiReference(this.transpile, this.assembly, options?.linkFormatter ?? ((t: TranspiledType) => `#${t.fqn}`), submodule);
+      const apiReference = new ApiReference(transpile, assembly, options?.linkFormatter ?? ((t: TranspiledType) => `#${t.fqn}`), submodule);
       documentation.section(apiReference.render());
     }
 
     return documentation;
   }
 
+  private addCleanupDirectory(directory: string) {
+    this.cleanupDirectories.push(directory);
+  }
+
+  /**
+   * Removes any internal working directories.
+   */
+  public async cleanup() {
+    for (const dir of this.cleanupDirectories) {
+      await fs.remove(dir);
+    }
```

Comment:
Maybe also remove the entry from `this.cleanupDirectories`? Or drop the list once it's done?